### ISSUE TITLE
ScalaJSDefined objects don't count as "plain objects"

### ISSUE THREAD

**Issue by jducoeur** [scala-js/scala-js#2006]
jQuery has a function $.isPlainObject, defined as:
```
var class2type = {};
var hasOwn = class2type.hasOwnProperty;
...
	isPlainObject: function( obj ) {
		// Not plain objects:
		// - Any object or value whose internal [[Class]] property is not "[object Object]"
		// - DOM nodes
		// - window
		if ( jQuery.type( obj ) !== "object" || obj.nodeType || jQuery.isWindow( obj ) ) {
			return false;
		}

		if ( obj.constructor &&
				!hasOwn.call( obj.constructor.prototype, "isPrototypeOf" ) ) {
			return false;
		}

		// If the function hasn't returned already, we're confident that
		// |obj| is a plain object, created by {} or constructed with new Object
		return true;
	}
```
In other words, something is a "plain object" iff *obj*.constructor.prototype.hasOwnProperty("isPrototypeOf").  As far as I can tell, this is used as a sanity-check, to make sure that you're not passing DOM objects and the like around as config objects.  (It seems idiosyncratic, but [does seem to match the standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/isPrototypeOf)).

The problem is that, by this definition, ScalaJSDefined objects don't count as "plain objects", because their constructor prototype doesn't include the "isPrototypeOf" function.  The result is that, if you use ScalaJSDefined to build configuration objects, they may fail to function (and indeed, break the library's usage entirely) because of this.

The extent of this problem isn't completely clear yet.  It definitely causes the [jsTree Widget](https://www.jstree.com/docs/config/), which I was using for my initial test of ScalaJSDefined, to fail completely.  On a quick code skim of Querki's -jsdeps, it looks like it would also cause the [Manifest library](https://github.com/jstayton/jquery-manifest) to fail.  The other libraries don't appear to use isPlainObject explicitly; I don't yet know whether jQuery's own internal use of the function (in half a dozen places) would cause issues.

**Comment by sjrd** [comment]
I understand this might be a problem for jQuery, but this behavior is nevertheless as designed. If jQuery insists on demanding "pain objects", then you can't use Scala.js-defined classes for that purpose.

**Comment by jducoeur** [comment]
Not totally surprised, and it may be insuperable, in which case just WontFix it; in that case, I'll just keep promulgating JSOptionBuilder for this purpose.  I'll just note that this is about 90% of my potential usage of ScalaJSDefined -- it's relatively uncommon for me to define JS literals for anything other than jQuery configuration objects, and they make up the vast bulk of most jQuery facades...

**Comment by sjrd** [comment]
Closing as "as-designed". It would be "fixed" by #2009, though; no need to keep two issues open.
